emscripten.py

```diff
@@ -1833,9 +1826,91 @@ def build_wasm(temp_files, infile, outfile, settings, DEBUG):
   return wast
 
 
-def create_metadata_wasm(metadata_raw, wast):
+def build_wasm_lld(temp_files, infile, outfile, settings, DEBUG):
+  lld = os.path.join(shared.LLVM_ROOT, 'ld.lld')
```

Comment:
this should probably be a constant `LLD` in `shared.py`, along the ones for `CLANG` etc., unless this is special somehow?

is ld.lld provided in clang builds normally? or would the user need to get it separately?

eventually we'd need to check if it exists and at least show a clear error explaining what to do if it's not.